### ISSUE TITLE
ArcGIS Runtime SDK for Java v100.13 - problem using latest MIL-STD-2525C stylx file

### ISSUE THREAD

**Issue by mmarionRH** [Esri/military-symbol-editor-addin-wpf#399]
When using the latest stylx file (https://www.arcgis.com/home/item.html?id=4e31ddc1f609432d98bd400f87f6b7bf) with ArcGIS Runtime SDK for Java v.100.13, the entities that use CIMMarkerPlacementAroundPolygon marker placement will cause an exception:

`java.util.concurrent.ExecutionException: com.esri.arcgisruntime.ArcGISRuntimeException: JSON parser invalid token.
      at com.esri.arcgisruntime.internal.concurrent.CoreTaskListenableFuture.get(CoreTaskListenableFuture.java:130)`

I found that the entities that have this issue were using another marker placement (CIMMarkerPlacementPolygonCenter) in previous stylx files.

Also, if you remove the “position” attribute from the entity’s Json script, the exception stops being thrown. Still, according to [documentation](https://pro.arcgis.com/en/pro-app/latest/sdk/api-reference/#topic29794.html), position attribute is expected and should work fine.

**Comment by csmoore** [comment]
I can definitely make sure the Runtime team is made aware of this issue.

I can look at the style JSON to find which MIL-symbols have this attribute - but if you have a SIDC -or- better yet a feature class that shows this issue that will help me writeup the repro case.

**Comment by mmarionRH** [comment]
Sure I can provide some sidc.

- Maneuver Lines : Airhead Line (G-G-SAA---_labels)
- Protection Areas : Mined Area (G-M-OFA---_labels)

If you look at these json, you'll see the following:

`"markerPlacement": {
    "type": "CIMMarkerPlacementAroundPolygon",
    "position": "Top",
    "offset": 10
},`

The "position" attribute seems to be the one to cause the exception (if you remove it, placement seems to work with a default value).

I've been comparing with a previous dictionary and the same symbols had this marker placement instead:

`"markerPlacement": {
    "type": "CIMMarkerPlacementPolygonCenter",
    "method": "OnPolygon",
    "offsetY": 15
},`

If you do replace the marker placement content with these previous values, the exception stops occurring.

Thank you

**Comment by mmarionRH** [comment]
I did a sample to showcase this issue (which I'm including here). In the sample, use the “Load & Show Entities That Are OK” button to use entities that do not cause the exception, and use the “Load & Show Entities That Have Issues” to trigger the exception.

To run the sample, update extension to .java, update the package and the path to the MIL-STD-2525C stylx file.

Hope it helps.

[CIMMarkerPlacementAroundPolygonIssueSample.txt](https://github.com/Esri/military-symbology/files/8383571/CIMMarkerPlacementAroundPolygonIssueSample.txt)

**Comment by csmoore** [comment]
Thanks for providing such a comprehensive sample showing this important issue and the exact cause. 

Sorry for the delay getting back. Because I have not used Java for 5 years, I apparently had a lot of catching up to do – it used to be quite easy to compile/run these samples, but now there seem to be lots of compile-time and runtime dependencies beyond just the JDK (gradle, JavaFX, API keys, etc.)

I was able to repro with your sample so I can enter an issue for this is our internal issue tracker (UPDATE: internal issue you can reference is runtime/java#6673) – you may want to also submit a Support ticket just to make sure it gets prioritized for next release.
 
A few of my observations (that you probably have already noticed as well):

•	I tested in 100.13.1 so still an issue there (screenshot below)
•	Reverting to the previous style version works around the issue – but obviously does not support new capabilities - like ordered-anchor points 
•	I looked at the style and noticed there are 54 tactical graphic symbols with that particular CIM tag (on label placement CIMs - list below)
•	I tried those symbols in latest Pro and they seemed to work there – so this CIM issue seems to be isolated to Runtime

I made a couple of minor updates to be able to package this as a complete repro case in order to submit: [JavaRTDRCrashRepro.zip](https://github.com/Esri/military-symbology/files/8400721/JavaRTDRCrashRepro.zip)

![image](https://user-images.githubusercontent.com/3090809/161335776-33d22332-d157-4825-9e8a-2a4f44e75782.png)

G-G-SAA---_labels
G-F-ATG---_labels
G-F-AZII--_labels
G-F-AZIR--_labels
G-F-AZXI--_labels
G-F-AZXR--_labels
G-F-AZCI--_labels
G-F-AZCR--_labels
G-F-AZFI--_labels
G-F-AZFR--_labels
G-F-ACDI--_labels
G-F-ACDR--_labels
G-F-ACDC--_labels
G-F-ACEI--_labels
G-F-ACER--_labels
G-F-ACEC--_labels
G-F-ACBI--_labels
G-F-ACBR--_labels
G-F-ACBC--_labels
G-F-ACVI--_labels
G-F-ACVR--_labels
G-F-ACVC--_labels
G-F-ACZI--_labels
G-F-ACZR--_labels
G-F-ACZC--_labels
G-F-AKBI--_labels
G-F-AKBR--_labels
G-F-AKBC--_labels
G-F-AKPI--_labels
G-F-AKPR--_labels
G-F-AKPC--_labels
G-M-OFS---_labels
G-G-PC----_labels
G-M-OFD---_labels
G-M-OFA---_labels
G-F-AZIC--_labels
G-F-AZXC--_labels
G-F-AZCC--_labels
G-F-AZFC--_labels
G-F-AZDC--_labels
G-F-AZDI--_labels
G-F-AZDR--_labels
G-F-AZSC--_labels
G-F-AZSI--_labels
G-F-AZSR--_labels
G-F-AZBC--_labels
G-F-AZBI--_labels
G-F-AZBR--_labels
G-F-AZVC--_labels
G-F-AZVI--_labels
G-F-AZVR--_labels
G-F-AZZC--_labels
G-F-AZZI--_labels
G-F-AZZR--_labels

Call stack:
```
java.util.concurrent.ExecutionException: com.esri.arcgisruntime.ArcGISRuntimeException: JSON parser invalid token.
        at com.esri.arcgisruntime.internal.concurrent.CoreTaskListenableFuture.get(CoreTaskListenableFuture.java:130)
        at com.example.app.CIMMarkerPlacementAroundPolygonIssueSample.createGraphic(CIMMarkerPlacementAroundPolygonIssueSample.java:236)
        at com.example.app.CIMMarkerPlacementAroundPolygonIssueSample.createEntitiesThatHaveIssues(CIMMarkerPlacementAroundPolygonIssueSample.java:209)
        at com.example.app.CIMMarkerPlacementAroundPolygonIssueSample.lambda$createButtons$4(CIMMarkerPlacementAroundPolygonIssueSample.java:153)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
```

**Comment by pMaske** [comment]
Thanks for reporting the issue. Fix is planned for upcoming release of ArcGIS Runtime SDK.

**Comment by csmoore** [comment]
I also confirmed the fix in the upcoming build using your sample - thanks again for reporting this issue and providing the great repro case:

![image](https://user-images.githubusercontent.com/3090809/163079864-5d8e8986-4678-4241-b17a-c9f7d5740ee8.png)

**Comment by mmarionRH** [comment]
Thank you to both of you for the good support.

**Comment by mmarionRH** [comment]
Hi @csmoore and @pMaske,

Now that release 100.14.0 is available, I tested my sample with it to see if the exception still appears, and it does.

![image](https://user-images.githubusercontent.com/58786169/169372920-3b99dcdc-d18e-4e61-8169-5ef5344bed20.png)

As you can see, I'm on version 100.14. Just to make sure I absolutely was on 100.14 and not on 100.13.1, I added a call to [ArcadeEvaluator](https://developers.arcgis.com/java/api-reference/reference/com/esri/arcgisruntime/arcade/ArcadeEvaluator.html)'s constructor, which would not compile on anything below 100.14 (that's the first log you see in the screenshot).

I see there's no new dictionary, so I am using [the latest one](https://www.arcgis.com/home/item.html?id=4e31ddc1f609432d98bd400f87f6b7bf). I also don't see any mention of this issue in the [release notes](https://developers.arcgis.com/java/reference/release-notes/100.14/) (but there's a lot of text, maybe I missed it).

Am I missing something? Thank you

**Comment by csmoore** [comment]
@mmarionRH - when I [tested above](https://github.com/Esri/military-symbology/issues/399#issuecomment-1097452543) on the daily build I just used the released dictionaries - but I do know that this daily test was very close to the release cut (+/- 5 days) - so it is possible it may not have made it - let me check/confirm ( probably with @pMaske ) what release this was expected to be in.

**Comment by pMaske** [comment]
@mmarionRH I re-tested  my .Net reproducer app ( that was basically a port of your Java app ) with released 100.14 binaries and the graphics render fine. Code snippet below has the codes that I am drawing.

![mil2525cRepro](https://user-images.githubusercontent.com/3878047/169394723-8875a3ed-c324-41fa-b4c1-f73b3810f336.JPG)

```
 pointCollection.Add(0, 0);
            pointCollection.Add(25, 0);
            pointCollection.Add(50, -20);
            createGraphic("GFGPSAA-------X", pointCollection, overlay1, ds);
            createGraphicInDictionaryRenderer("GFGPSAA-------X", pointCollection, overlay2);

            pointCollection.Clear();
            pointCollection.Add(0, -25);
            pointCollection.Add(50, -45);
            pointCollection.Add(25, -25);
            createGraphic("GFMPOFA-------X", pointCollection, overlay1, ds);
            createGraphicInDictionaryRenderer("GFMPOFA-------X", pointCollection, overlay2);

            pointCollection.Clear();
            pointCollection.Add(0, 10);
            pointCollection.Add(50, 30);
            pointCollection.Add(25, 10);
            createGraphic("G*GPGAA---****X", pointCollection, overlay1, ds);
            createGraphicInDictionaryRenderer("G*GPGAA---****X", pointCollection, overlay2);

            pointCollection.Clear();
            pointCollection.Add(0, 35);
            pointCollection.Add(50, 55);
            pointCollection.Add(25, 35);
            createGraphic("G*GPPM----****X", pointCollection, overlay1, ds);
            createGraphicInDictionaryRenderer("G*GPPM----****X", pointCollection, overlay2);
```

**Comment by csmoore** [comment]
@mmarionRH - I also reran on your [modified sample I added above](https://github.com/Esri/military-symbology/issues/399#issuecomment-1086297238) with 100.14 version (that gets downloaded by gradle from the artifact repo).

So not sure if this is configuration issue or just a new different issue you are finding - can you confirm that we are using the same sample about to test just to rule out a configuration issue?

Sample I used: [JavaRTDRCrashRepro.zip](https://github.com/Esri/military-symbology/files/8734049/JavaRTDRCrashRepro.zip)

Note: Only change to sample above:
`build.gradle`
```
ext {
    arcgisVersion = '100.14.0'
}
```

![image](https://user-images.githubusercontent.com/3090809/169397791-c31b4e2a-5408-4f85-92d9-8fdc471166e6.png)

**Comment by mmarionRH** [comment]
Hi @csmoore ,

Here's what I did (and still can't see the fix):

- Compared my sample's code with yours, just to see what you did different and apply it in my sample.
- Ran your sample using my setup (I am using Maven).
- Ran your sample as-is (Gradle).

Here's a screenshot from the Gradle build:

![image](https://user-images.githubusercontent.com/58786169/169596201-19996ffc-e7d7-41e7-9823-2c4b6d4078f6.png)

As you can see, I added a few labels to show the JDK and JavaFx versions. Both seem to be as required in the [release notes](https://developers.arcgis.com/java/reference/system-requirements/).

I've been turning this upside down for a few hours now. The only thing I couldn't validate is if my dictionary is different than the one you have here: `private static final String STYLX_FILE = "C:/MyFiles/Installs/arcgis-runtime-sdk-java-100.13.1/Samples/display-a-map-solution/mil2525c.stylx";`. As I said in my previous comment, I am using [this dictionary](https://www.arcgis.com/home/item.html?id=4e31ddc1f609432d98bd400f87f6b7bf).

**Comment by pMaske** [comment]
@mmarionRH in my test, I downloaded the same stylx you are referring to i.e. https://www.arcgis.com/home/item.html?id=4e31ddc1f609432d98bd400f87f6b7bf and still don't see this problem with 100.14.

**Comment by csmoore** [comment]
@mmarionRH @pMaske - Thanks for trying this out - I also just verified that I was using the latest style from: https://www.arcgis.com/home/item.html?id=4e31ddc1f609432d98bd400f87f6b7bf  - sorry for the bad link in the code - it was running with the [stylx, size 14,246, that was included in the zip above](https://github.com/Esri/military-symbology/issues/399#issuecomment-1132178993)

It definitely sounds like we are getting different results with the same sample/style. I'm really not sure what the underlying configuration issue is. The next step may be opening a trouble ticket (if there isn't already one open) and having a support tech take a look at.

![image](https://user-images.githubusercontent.com/3090809/169830132-a9b52ea3-881f-41aa-b09a-266fb8ceb31b.png)


```
C:\MyFiles\Installs\SymbolTest100.14\JavaRTDRCrashRepro>gradle run

> Configure project :
Project : => no module-info.java found

> Task :compileJava
C:\MyFiles\Installs\SymbolTest100.14\JavaRTDRCrashRepro\src\main\java\com\example\app\CIMMarkerPlacementAroundPolygonIssueSample.java:150: warning: [removal] getAPIVersion() in ArcGISRuntimeEnvironment has been deprecated and marked for removal
        Label version = new Label(String.format("Version -> %s (%s)", ArcGISRuntimeEnvironment.getAPIVersion(), ArcGISRuntimeEnvironment.getLicense().getLicenseLevel()));
                                                                                              ^
1 warning

> Task :run
Initializing...
Java version : 11.0.11 (AdoptOpenJDK) amd64
<==========---> 83% EXECUTING [17s]
> :run
```

**Comment by mmarionRH** [comment]
Hi @pMaske & @csmoore,

Seems like I was in a weird Frankenstein state in-between 100.13.1 and 100.14. Now that I found how to resolve that, everything works fine:

![image](https://user-images.githubusercontent.com/58786169/170053888-858843c5-fe71-499a-8569-7d5dba60b065.png)

Thank you for your patience on this, I appreciate it. Have a good day!

**Comment by csmoore** [comment]
@mmarionRH - glad you got things working - thanks for letting us know. As someone who tried to run Java for the first time in 5+ years above, I can certainly sympathize with configuration issues (starting with [several different JDK/JREs](https://www.baeldung.com/oracle-jdk-vs-openjdk))  

We appreciate the help solving this issue in time to fix for 100.14.
